### ISSUE TITLE
CORS issues with Latest Firefox

### ISSUE THREAD

**Issue by KameshDevarakonda** [google/shaka-player#1476]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**: Yes

**What version of Shaka Player are you using?**: v2.4.1-uncompiled

**Can you reproduce the issue with our latest release version?**: yes

**Can you reproduce the issue with the latest code from `master`?**: 

**Are you using the demo app or your own custom app?**: Demo app

**If custom app, can you reproduce the issue using our demo app?**:

**What browser and OS are you using?**: macOS High Sierra, Linux debian
Firefox 62

**What are the manifest and license server URIs?**:
https://docs.google.com/document/d/1gWfFtrnLkFHaEMDcLWgNmx-r5Hm1oA_8bUjuwegW_IE/edit?usp=sharing

**What did you do?**
More details in the doc.

**What did you expect to happen?**
Smooth Playback

**What actually happened?**
CORS issues with redirects.

**Comment by joeyparrish** [comment]
I think this is a duplicate of #666, which is a Firefox bug being discussed here: https://bugzilla.mozilla.org/show_bug.cgi?id=1312864

CORS should work with redirects, but Firefox has not caught up to the spec.

I'm going to close this as a duplicate now, but please feel free to subscribe to #666 or go to their buganizer and let Mozilla know that this issue is important to you.

**Comment by KameshDevarakonda** [comment]
Please correct me if I am wrong but the bug you linked to specifies that redirects are not followed if it requires preflight request.

The workflow I described in the doc seems to point to "simple" CORS requests (that don't require preflight). 

Apart from that as I documented in the doc, Firefox and Dash.js combination works flawless. Also Shaka Player works flawlessly in the older version of Firefox (52 ESR).

**Comment by joeyparrish** [comment]
Then it's possible that I was mistaken.  We will try to reproduce your issue and let you know what we find.

**Comment by joeyparrish** [comment]
Sanitized details from the doc linked in the original post:

-----

Manifest generation involves private tools (shared with us for repro).
Browser: Firefox 62.0b3

A request for media segments resulted in a redirect.  The redirected URL was then subsequently blocked by the browser due to CORS ("Same Origin Policy").

Error logged by Firefox: `Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at ...
Reason: CORS header ‘Access-Control-Allow-Origin’ does not match ‘https://shaka-player-demo.appspot.com’`

This results in `NETWORK.HTTP_ERROR` at the Player level.

"Setting origin value to null in the redirect seems to solve cors issue."

**Comment by joeyparrish** [comment]
I am unable to reproduce the issue in Firefox 62.0b3.  Did you apply the workaround you mentioned in your doc?

**Comment by joeyparrish** [comment]
I also tried the latest Firefox beta (62.0b7) and couldn't reproduce in that, either.

**Comment by joeyparrish** [comment]
The other thing I'm confused about is the workaround itself.  You said "Setting origin value to null in the redirect", but origin is a request header, not a response header.  Can you clarify that?

**Comment by KameshDevarakonda** [comment]
I can still reproduce the issue. Did you clear the cache before running the test?

Regarding second point, when I used the dash.js player "the origin request header is being set as null in the redirect"

![screen shot 2018-07-12 at 5 12 19 pm](https://user-images.githubusercontent.com/37383885/42659688-07a9845e-85f7-11e8-898f-065b38389bbe.png)
![screen shot 2018-07-12 at 5 10 44 pm](https://user-images.githubusercontent.com/37383885/42659701-11874696-85f7-11e8-9005-418075f62eb6.png)

**Comment by joeyparrish** [comment]
@KameshDevarakonda, when this bug occurs, do you have any console logs in Firefox about the ServiceWorker?  In a similar issue (#1498), I found this error log in the console:

`A ServiceWorker passed a promise to FetchEvent.respondWith() that rejected with ‘TypeError: NetworkError when attempting to fetch resource.’.`

If you see this, too, then I can consider the two issues to be the same, and work on a fix via #1498, which is easier for me to reproduce.

**Comment by KameshDevarakonda** [comment]
Yep.. I can see exact same message : A ServiceWorker passed a promise to FetchEvent.respondWith() that rejected with ‘TypeError: NetworkError when attempting to fetch resource.’.

`16:06:09.503 Unload latency: 0.015 player.js:706:5
16:06:09.604 notifySegments: maxSegmentDuration=10.00780045351474
presentation_timeline.js:192:3
16:06:09.626 notifyNewSegmentDuration: maxSegmentDuration=10.00780045351474 presentation_timeline.js:212:3
16:06:09.626 Found variant with audio and video content, so filtering all periods. player.js:607:5
16:06:09.628 codecs avc1-mp4a avg bandwidth 10607967.42857143 player.js:900:5
16:06:09.629 onChooseStreams_ 
Object { startTime: 0, textStreams: [], variants: (1) […] }
player.js:2839:3
16:06:09.630 Choosing new streams after period changed player.js:2851:3
16:06:09.631 onChooseStreams_, variants and text streams:  
Array [ {…} ]
 
Array []
player.js:2871:3
16:06:09.632 Bandwidth ranges: 3.411 3.591 Infinity simple_abr_manager.js:142:5
16:06:09.632 onChooseStreams_, chosen= 
Object { id: 3, language: "und", primary: false, audio: {…}, video: {…}, bandwidth: 3411000, drmInfos: [], allowedByApplication: true, allowedByKeySystem: true }
 null player.js:2877:3
16:06:09.635 (all) setting up Stream 1 streaming_engine.js:1018:7
16:06:09.635 (all) setting up Stream 2 streaming_engine.js:1018:7
16:06:09.635 (all) setup Stream 1 streaming_engine.js:1036:9
16:06:09.635 (all) setup Stream 2 streaming_engine.js:1036:9
16:06:09.636 (audio:1) updating in 0 seconds streaming_engine.js:2205:3
16:06:09.636 (video:2) updating in 0 seconds streaming_engine.js:2205:3
16:06:09.636 init: completed initial Stream setup streaming_engine.js:423:5
16:06:09.641 (audio:1) timeNeeded=0 streaming_engine.js:1160:3
16:06:09.642 (audio:1) update_: playheadTime=0 bufferedAhead=0 streaming_engine.js:1171:3
16:06:09.642 (audio:1) next position unknown: nothing buffered streaming_engine.js:1346:5
16:06:09.642 (audio:1) looking up segment: presentationTime=0 currentPeriod.startTime=0 streaming_engine.js:1387:3
16:06:09.642 (audio:1) segment does not exist: currentPeriod.startTime=0 position=0 streaming_engine.js:1423:5
16:06:09.643 (audio:1) fetchAndAppend_: playheadTime=0 currentPeriod.startTime=0 reference.position=1 reference.startTime=0 reference.endTime=10.00780045351474 streaming_engine.js:1467:3
16:06:09.643 (audio:1) setting timestamp offset to 0 streaming_engine.js:1690:3
16:06:09.643 (audio:1) setting appstart window start to 0 streaming_engine.js:1691:3
16:06:09.643 (audio:1) setting append window end to 30.093 streaming_engine.js:1693:3
16:06:09.644 (audio:1) fetching init segment streaming_engine.js:1703:3
16:06:09.645 fetching: reference=[object Object] streaming_engine.js:2150:3
16:06:09.645 (audio:1) fetching segment streaming_engine.js:1503:3
16:06:09.649 fetching: reference=[object Object] streaming_engine.js:2150:3
16:06:09.654 (video:2) timeNeeded=0 streaming_engine.js:1160:3
16:06:09.654 (video:2) update_: playheadTime=0 bufferedAhead=0 streaming_engine.js:1171:3
16:06:09.654 (video:2) next position unknown: nothing buffered streaming_engine.js:1346:5
16:06:09.655 (video:2) looking up segment: presentationTime=0 currentPeriod.startTime=0 streaming_engine.js:1387:3
16:06:09.655 (video:2) segment does not exist: currentPeriod.startTime=0 position=0 streaming_engine.js:1423:5
16:06:09.655 (video:2) fetchAndAppend_: playheadTime=0 currentPeriod.startTime=0 reference.position=1 reference.startTime=0 reference.endTime=5.005 streaming_engine.js:1467:3
16:06:09.656 (video:2) setting timestamp offset to 0 streaming_engine.js:1690:3
16:06:09.656 (video:2) setting appstart window start to 0 streaming_engine.js:1691:3
16:06:09.656 (video:2) setting append window end to 30.093 streaming_engine.js:1693:3
16:06:09.657 (video:2) fetching init segment streaming_engine.js:1703:3
16:06:09.657 fetching: reference=[object Object] streaming_engine.js:2150:3
16:06:09.657 (video:2) fetching segment streaming_engine.js:1503:3
16:06:09.658 fetching: reference=[object Object] streaming_engine.js:2150:3
16:06:09.976 Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://truman-nightly.sandbox.google.com/segments/redirect/c/init.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26ctier%3DL%26clen%3D478945%26mime%3Daudio%2Fmp4%26xtags%3Dac%3Daac%3Aach%3D2%3Aasr%3D44100%3Aatbr%3D128%26cmbypass%3Dyes%26ratebypass%3Dyes%26acao%3Dyes%26gir%3Dyes%26lmt%3D1532061054141045%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Cctier%2Cclen%2Cmime%2Cxtags%2Ccmbypass%2Cratebypass%2Cacao%2Cgir%2Clmt%26signature%3D5350291D739FFA03F114F99B70C0709439249583.103BC4499B0D879A7611EE7B350D50C314EB0E1A%26key%3Dck2. (Reason: CORS request did not succeed).
16:06:09.976 Failed to load ‘https://truman-nightly.sandbox.google.com/segments/redirect/c/init.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26ctier%3DL%26clen%3D478945%26mime%3Daudio%2Fmp4%26xtags%3Dac%3Daac%3Aach%3D2%3Aasr%3D44100%3Aatbr%3D128%26cmbypass%3Dyes%26ratebypass%3Dyes%26acao%3Dyes%26gir%3Dyes%26lmt%3D1532061054141045%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Cctier%2Cclen%2Cmime%2Cxtags%2Ccmbypass%2Cratebypass%2Cacao%2Cgir%2Clmt%26signature%3D5350291D739FFA03F114F99B70C0709439249583.103BC4499B0D879A7611EE7B350D50C314EB0E1A%26key%3Dck2’. A ServiceWorker passed a promise to FetchEvent.respondWith() that rejected with ‘TypeError: NetworkError when attempting to fetch resource.’.
16:06:09.988 Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://truman-nightly.sandbox.google.com/segments/redirect/c/init.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26cmbypass%3Dyes%26ratebypass%3Dyes%26gir%3Dyes%26clen%3D12328325%26acao%3Dyes%26ctier%3DL%26lmt%3D1532061116278872%26mime%3Dvideo%2Fmp4%26xtags%3Dvc%3Davc%3Avccf%3Dmain%3Avsz%3D1920x1080%3Avtbr%3D4400%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Ccmbypass%2Cratebypass%2Cgir%2Cclen%2Cacao%2Cctier%2Clmt%2Cmime%2Cxtags%26signature%3D1A8F5A22887555328AC1D2E1FB872CC11CE48479.5448645BCC34CB63753060DF1DE782F7C4017009%26key%3Dck2. (Reason: CORS request did not succeed).
16:06:09.988 Failed to load ‘https://truman-nightly.sandbox.google.com/segments/redirect/c/init.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26cmbypass%3Dyes%26ratebypass%3Dyes%26gir%3Dyes%26clen%3D12328325%26acao%3Dyes%26ctier%3DL%26lmt%3D1532061116278872%26mime%3Dvideo%2Fmp4%26xtags%3Dvc%3Davc%3Avccf%3Dmain%3Avsz%3D1920x1080%3Avtbr%3D4400%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Ccmbypass%2Cratebypass%2Cgir%2Cclen%2Cacao%2Cctier%2Clmt%2Cmime%2Cxtags%26signature%3D1A8F5A22887555328AC1D2E1FB872CC11CE48479.5448645BCC34CB63753060DF1DE782F7C4017009%26key%3Dck2’. A ServiceWorker passed a promise to FetchEvent.respondWith() that rejected with ‘TypeError: NetworkError when attempting to fetch resource.’.
16:06:09.998 Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://truman-nightly.sandbox.google.com/segments/redirect/c/1.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26cmbypass%3Dyes%26ratebypass%3Dyes%26gir%3Dyes%26clen%3D12328325%26acao%3Dyes%26ctier%3DL%26lmt%3D1532061116278872%26mime%3Dvideo%2Fmp4%26xtags%3Dvc%3Davc%3Avccf%3Dmain%3Avsz%3D1920x1080%3Avtbr%3D4400%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Ccmbypass%2Cratebypass%2Cgir%2Cclen%2Cacao%2Cctier%2Clmt%2Cmime%2Cxtags%26signature%3D1A8F5A22887555328AC1D2E1FB872CC11CE48479.5448645BCC34CB63753060DF1DE782F7C4017009%26key%3Dck2. (Reason: CORS request did not succeed).
16:06:09.998 Failed to load ‘https://truman-nightly.sandbox.google.com/segments/redirect/c/1.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26cmbypass%3Dyes%26ratebypass%3Dyes%26gir%3Dyes%26clen%3D12328325%26acao%3Dyes%26ctier%3DL%26lmt%3D1532061116278872%26mime%3Dvideo%2Fmp4%26xtags%3Dvc%3Davc%3Avccf%3Dmain%3Avsz%3D1920x1080%3Avtbr%3D4400%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Ccmbypass%2Cratebypass%2Cgir%2Cclen%2Cacao%2Cctier%2Clmt%2Cmime%2Cxtags%26signature%3D1A8F5A22887555328AC1D2E1FB872CC11CE48479.5448645BCC34CB63753060DF1DE782F7C4017009%26key%3Dck2’. A ServiceWorker passed a promise to FetchEvent.respondWith() that rejected with ‘TypeError: NetworkError when attempting to fetch resource.’.
16:06:10.012 Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://truman-nightly.sandbox.google.com/segments/redirect/c/1.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26ctier%3DL%26clen%3D478945%26mime%3Daudio%2Fmp4%26xtags%3Dac%3Daac%3Aach%3D2%3Aasr%3D44100%3Aatbr%3D128%26cmbypass%3Dyes%26ratebypass%3Dyes%26acao%3Dyes%26gir%3Dyes%26lmt%3D1532061054141045%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Cctier%2Cclen%2Cmime%2Cxtags%2Ccmbypass%2Cratebypass%2Cacao%2Cgir%2Clmt%26signature%3D5350291D739FFA03F114F99B70C0709439249583.103BC4499B0D879A7611EE7B350D50C314EB0E1A%26key%3Dck2. (Reason: CORS request did not succeed).
16:06:10.012 Failed to load ‘https://truman-nightly.sandbox.google.com/segments/redirect/c/1.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26ctier%3DL%26clen%3D478945%26mime%3Daudio%2Fmp4%26xtags%3Dac%3Daac%3Aach%3D2%3Aasr%3D44100%3Aatbr%3D128%26cmbypass%3Dyes%26ratebypass%3Dyes%26acao%3Dyes%26gir%3Dyes%26lmt%3D1532061054141045%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Cctier%2Cclen%2Cmime%2Cxtags%2Ccmbypass%2Cratebypass%2Cacao%2Cgir%2Clmt%26signature%3D5350291D739FFA03F114F99B70C0709439249583.103BC4499B0D879A7611EE7B350D50C314EB0E1A%26key%3Dck2’. A ServiceWorker passed a promise to FetchEvent.respondWith() that rejected with ‘TypeError: NetworkError when attempting to fetch resource.’.
16:06:10.608 (audio:1) failed fetch and append: code=1002 streaming_engine.js:1554:7
16:06:10.607 Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://truman-nightly.sandbox.google.com/segments/redirect/c/init.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26ctier%3DL%26clen%3D478945%26mime%3Daudio%2Fmp4%26xtags%3Dac%3Daac%3Aach%3D2%3Aasr%3D44100%3Aatbr%3D128%26cmbypass%3Dyes%26ratebypass%3Dyes%26acao%3Dyes%26gir%3Dyes%26lmt%3D1532061054141045%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Cctier%2Cclen%2Cmime%2Cxtags%2Ccmbypass%2Cratebypass%2Cacao%2Cgir%2Clmt%26signature%3D5350291D739FFA03F114F99B70C0709439249583.103BC4499B0D879A7611EE7B350D50C314EB0E1A%26key%3Dck2. (Reason: CORS request did not succeed).
16:06:10.607 Failed to load ‘https://truman-nightly.sandbox.google.com/segments/redirect/c/init.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26ctier%3DL%26clen%3D478945%26mime%3Daudio%2Fmp4%26xtags%3Dac%3Daac%3Aach%3D2%3Aasr%3D44100%3Aatbr%3D128%26cmbypass%3Dyes%26ratebypass%3Dyes%26acao%3Dyes%26gir%3Dyes%26lmt%3D1532061054141045%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Cctier%2Cclen%2Cmime%2Cxtags%2Ccmbypass%2Cratebypass%2Cacao%2Cgir%2Clmt%26signature%3D5350291D739FFA03F114F99B70C0709439249583.103BC4499B0D879A7611EE7B350D50C314EB0E1A%26key%3Dck2’. A ServiceWorker passed a promise to FetchEvent.respondWith() that rejected with ‘TypeError: NetworkError when attempting to fetch resource.’.
16:06:10.736 (video:2) failed fetch and append: code=1002 streaming_engine.js:1554:7
16:06:10.735 Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://truman-nightly.sandbox.google.com/segments/redirect/c/init.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26cmbypass%3Dyes%26ratebypass%3Dyes%26gir%3Dyes%26clen%3D12328325%26acao%3Dyes%26ctier%3DL%26lmt%3D1532061116278872%26mime%3Dvideo%2Fmp4%26xtags%3Dvc%3Davc%3Avccf%3Dmain%3Avsz%3D1920x1080%3Avtbr%3D4400%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Ccmbypass%2Cratebypass%2Cgir%2Cclen%2Cacao%2Cctier%2Clmt%2Cmime%2Cxtags%26signature%3D1A8F5A22887555328AC1D2E1FB872CC11CE48479.5448645BCC34CB63753060DF1DE782F7C4017009%26key%3Dck2. (Reason: CORS request did not succeed).
16:06:10.735 Failed to load ‘https://truman-nightly.sandbox.google.com/segments/redirect/c/init.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26cmbypass%3Dyes%26ratebypass%3Dyes%26gir%3Dyes%26clen%3D12328325%26acao%3Dyes%26ctier%3DL%26lmt%3D1532061116278872%26mime%3Dvideo%2Fmp4%26xtags%3Dvc%3Davc%3Avccf%3Dmain%3Avsz%3D1920x1080%3Avtbr%3D4400%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Ccmbypass%2Cratebypass%2Cgir%2Cclen%2Cacao%2Cctier%2Clmt%2Cmime%2Cxtags%26signature%3D1A8F5A22887555328AC1D2E1FB872CC11CE48479.5448645BCC34CB63753060DF1DE782F7C4017009%26key%3Dck2’. A ServiceWorker passed a promise to FetchEvent.respondWith() that rejected with ‘TypeError: NetworkError when attempting to fetch resource.’.
16:06:10.893 Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://truman-nightly.sandbox.google.com/segments/redirect/c/1.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26ctier%3DL%26clen%3D478945%26mime%3Daudio%2Fmp4%26xtags%3Dac%3Daac%3Aach%3D2%3Aasr%3D44100%3Aatbr%3D128%26cmbypass%3Dyes%26ratebypass%3Dyes%26acao%3Dyes%26gir%3Dyes%26lmt%3D1532061054141045%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Cctier%2Cclen%2Cmime%2Cxtags%2Ccmbypass%2Cratebypass%2Cacao%2Cgir%2Clmt%26signature%3D5350291D739FFA03F114F99B70C0709439249583.103BC4499B0D879A7611EE7B350D50C314EB0E1A%26key%3Dck2. (Reason: CORS request did not succeed).
16:06:10.893 Failed to load ‘https://truman-nightly.sandbox.google.com/segments/redirect/c/1.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26ctier%3DL%26clen%3D478945%26mime%3Daudio%2Fmp4%26xtags%3Dac%3Daac%3Aach%3D2%3Aasr%3D44100%3Aatbr%3D128%26cmbypass%3Dyes%26ratebypass%3Dyes%26acao%3Dyes%26gir%3Dyes%26lmt%3D1532061054141045%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Cctier%2Cclen%2Cmime%2Cxtags%2Ccmbypass%2Cratebypass%2Cacao%2Cgir%2Clmt%26signature%3D5350291D739FFA03F114F99B70C0709439249583.103BC4499B0D879A7611EE7B350D50C314EB0E1A%26key%3Dck2’. A ServiceWorker passed a promise to FetchEvent.respondWith() that rejected with ‘TypeError: NetworkError when attempting to fetch resource.’.
16:06:11.502 Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://truman-nightly.sandbox.google.com/segments/redirect/c/1.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26cmbypass%3Dyes%26ratebypass%3Dyes%26gir%3Dyes%26clen%3D12328325%26acao%3Dyes%26ctier%3DL%26lmt%3D1532061116278872%26mime%3Dvideo%2Fmp4%26xtags%3Dvc%3Davc%3Avccf%3Dmain%3Avsz%3D1920x1080%3Avtbr%3D4400%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Ccmbypass%2Cratebypass%2Cgir%2Cclen%2Cacao%2Cctier%2Clmt%2Cmime%2Cxtags%26signature%3D1A8F5A22887555328AC1D2E1FB872CC11CE48479.5448645BCC34CB63753060DF1DE782F7C4017009%26key%3Dck2. (Reason: CORS request did not succeed).
16:06:11.502 Failed to load ‘https://truman-nightly.sandbox.google.com/segments/redirect/c/1.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26cmbypass%3Dyes%26ratebypass%3Dyes%26gir%3Dyes%26clen%3D12328325%26acao%3Dyes%26ctier%3DL%26lmt%3D1532061116278872%26mime%3Dvideo%2Fmp4%26xtags%3Dvc%3Davc%3Avccf%3Dmain%3Avsz%3D1920x1080%3Avtbr%3D4400%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Ccmbypass%2Cratebypass%2Cgir%2Cclen%2Cacao%2Cctier%2Clmt%2Cmime%2Cxtags%26signature%3D1A8F5A22887555328AC1D2E1FB872CC11CE48479.5448645BCC34CB63753060DF1DE782F7C4017009%26key%3Dck2’. A ServiceWorker passed a promise to FetchEvent.respondWith() that rejected with ‘TypeError: NetworkError when attempting to fetch resource.’.
16:06:11.647 Player error 
Object { severity: 2, category: 1, code: 1002, data: (3) […], handled: false, message: "Shaka Error NETWORK.HTTP_ERROR (https://truman-nightly.sandbox.google.com/segments/redirect/c/init.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26ctier%3DL%26clen%3D478945%26mime%3Daudio%2Fmp4%26xtags%3Dac%3Daac%3Aach%3D2%3Aasr%3D44100%3Aatbr%3D128%26cmbypass%3Dyes%26ratebypass%3Dyes%26acao%3Dyes%26gir%3Dyes%26lmt%3D1532061054141045%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Cctier%2Cclen%2Cmime%2Cxtags%2Ccmbypass%2Cratebypass%2Cacao%2Cgir%2Clmt%26signature%3D5350291D739FFA03F114F99B70C0709439249583.103BC4499B0D879A7611EE7B350D50C314EB0E1A%26key%3Dck2,TypeError: NetworkError when attempting to fetch resource.,1)", stack: "shaka.util.Error@https://shaka-player-demo.appspot.com/lib/util/error.js:89:13\nshaka.net.HttpFetchPlugin/promise<@https://shaka-player-demo.appspot.com/lib/net/http_fetch_plugin.js:100:29\npromise callback*shaka.net.HttpFetchPlugin@https://shaka-player-demo.appspot.com/lib/net/http_fetch_plugin.js:69:17\nshaka.net.NetworkingEngine.prototype.send_/sendOperation<@https://shaka-player-demo.appspot.com/lib/net/networking_engine.js:473:21\nwrapChainCallback_@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:234:17\nchain/<@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:195:15\npromise callback*chain@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:175:5\nshaka.net.NetworkingEngine.prototype.send_@https://shaka-player-demo.appspot.com/lib/net/networking_engine.js:467:23\nshaka.net.NetworkingEngine.prototype.send_/sendOperation<@https://shaka-player-demo.appspot.com/lib/net/networking_engine.js:507:14\nwrapChainCallback_@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:234:17\nchain/<@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:208:15\npromise callback*chain@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:175:5\nshaka.net.NetworkingEngine.prototype.send_@https://shaka-player-demo.appspot.com/lib/net/networking_engine.js:467:23\nshaka.net.NetworkingEngine.prototype.makeRequestWithRetry_@https://shaka-player-demo.appspot.com/lib/net/networking_engine.js:416:10\nshaka.net.NetworkingEngine.prototype.request/requestOperation<@https://shaka-player-demo.appspot.com/lib/net/networking_engine.js:329:13\nwrapChainCallback_@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:234:17\nchain/<@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:195:15\npromise callback*chain@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:175:5\nshaka.net.NetworkingEngine.prototype.request@https://shaka-player-demo.appspot.com/lib/net/networking_engine.js:328:26\nshaka.media.StreamingEngine.prototype.fetch_@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:2151:12\nshaka.media.StreamingEngine.prototype.initSourceBuffer_@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:1704:19\nshaka.media.StreamingEngine.prototype.fetchAndAppend_@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:1494:26\nshaka.media.StreamingEngine.prototype.update_@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:1246:3\nshaka.media.StreamingEngine.prototype.onUpdate_@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:1102:17\nsetTimeout handler*shaka.media.StreamingEngine.prototype.scheduleUpdate_@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:2208:28\nshaka.media.StreamingEngine.prototype.initStreams_/<@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:884:9\npromise callback*shaka.media.StreamingEngine.prototype.initStreams_@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:864:10\nshaka.media.StreamingEngine.prototype.init@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:418:10\nshaka.Player.prototype.load@https://shaka-player-demo.appspot.com/lib/player.js:779:11\nAsync*shakaDemo.load/<@https://shaka-player-demo.appspot.com/demo/asset_section.js:313:12\npromise callback*shakaDemo.load@https://shaka-player-demo.appspot.com/demo/asset_section.js:311:3\nEventListener.handleEvent*shakaDemo.setupAssets_@https://shaka-player-demo.appspot.com/demo/asset_section.js:111:3\nshakaDemo.init/<@https://shaka-player-demo.appspot.com/demo/main.js:203:24\npromise callback*shakaDemo.init@https://shaka-player-demo.appspot.com/demo/main.js:185:5\nEventListener.handleEvent*@https://shaka-player-demo.appspot.com/demo/main.js:742:5\n" }
main.js:687:3
shakaDemo.onError_
https://shaka-player-demo.appspot.com/demo/main.js:687:3
shakaDemo.onErrorEvent_
https://shaka-player-demo.appspot.com/demo/main.js:678:3
shaka.util.FakeEventTarget.prototype.dispatchEvent
https://shaka-player-demo.appspot.com/lib/util/fake_event_target.js:117:9
shaka.cast.CastProxy.prototype.playerProxyLocalEvent_
https://shaka-player-demo.appspot.com/lib/cast/cast_proxy.js:579:3
shaka.util.FakeEventTarget.prototype.dispatchEvent
https://shaka-player-demo.appspot.com/lib/util/fake_event_target.js:117:9
shaka.Player.prototype.onError_
https://shaka-player-demo.appspot.com/lib/player.js:3066:3
shaka.media.StreamingEngine.prototype.handleStreamingError_/<
https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:2244:5
shaka.media.StreamingEngine.prototype.handleStreamingError_
https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:2238:3
shaka.media.StreamingEngine.prototype.fetchAndAppend_/<
https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:1558:7
shaka.media.StreamingEngine.prototype.fetchAndAppend_
https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:1506:3
shaka.media.StreamingEngine.prototype.update_
https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:1246:3
shaka.media.StreamingEngine.prototype.onUpdate_
https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:1102:17
shaka.media.StreamingEngine.prototype.scheduleUpdate_
https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:2208:28
shaka.media.StreamingEngine.prototype.initStreams_/<
https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:884:9
(Async: promise callback) shaka.media.StreamingEngine.prototype.initStreams_
https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:864:10
shaka.media.StreamingEngine.prototype.init
https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:418:10
shaka.Player.prototype.load
https://shaka-player-demo.appspot.com/lib/player.js:779:11
shakaDemo.load/<
https://shaka-player-demo.appspot.com/demo/asset_section.js:313:12
(Async: promise callback) shakaDemo.load
https://shaka-player-demo.appspot.com/demo/asset_section.js:311:3
(Async: EventListener.handleEvent) shakaDemo.setupAssets_
https://shaka-player-demo.appspot.com/demo/asset_section.js:111:3
shakaDemo.init/<
https://shaka-player-demo.appspot.com/demo/main.js:203:24
(Async: promise callback) shakaDemo.init
https://shaka-player-demo.appspot.com/demo/main.js:185:5
(Async: EventListener.handleEvent) <anonymous>
https://shaka-player-demo.appspot.com/demo/main.js:742:5
16:06:12.146 Player error 
Object { severity: 2, category: 1, code: 1002, data: (3) […], handled: false, message: "Shaka Error NETWORK.HTTP_ERROR (https://truman-nightly.sandbox.google.com/segments/redirect/c/init.mp4?orig=https%3A%2F%2Fr2---atl14s48.googlevideo.com%2Fvideoplayback%3Fid%3Df773b756f6a4ed78%26itag%3D340%26source%3Ddclk_video_ads%26cmbypass%3Dyes%26ratebypass%3Dyes%26gir%3Dyes%26clen%3D12328325%26acao%3Dyes%26ctier%3DL%26lmt%3D1532061116278872%26mime%3Dvideo%2Fmp4%26xtags%3Dvc%3Davc%3Avccf%3Dmain%3Avsz%3D1920x1080%3Avtbr%3D4400%26ip%3D0.0.0.0%26ipbits%3D0%26expire%3D1532721898%26sparams%3Dip%2Cipbits%2Cexpire%2Cid%2Citag%2Csource%2Ccmbypass%2Cratebypass%2Cgir%2Cclen%2Cacao%2Cctier%2Clmt%2Cmime%2Cxtags%26signature%3D1A8F5A22887555328AC1D2E1FB872CC11CE48479.5448645BCC34CB63753060DF1DE782F7C4017009%26key%3Dck2,TypeError: NetworkError when attempting to fetch resource.,1)", stack: "shaka.util.Error@https://shaka-player-demo.appspot.com/lib/util/error.js:89:13\nshaka.net.HttpFetchPlugin/promise<@https://shaka-player-demo.appspot.com/lib/net/http_fetch_plugin.js:100:29\npromise callback*shaka.net.HttpFetchPlugin@https://shaka-player-demo.appspot.com/lib/net/http_fetch_plugin.js:69:17\nshaka.net.NetworkingEngine.prototype.send_/sendOperation<@https://shaka-player-demo.appspot.com/lib/net/networking_engine.js:473:21\nwrapChainCallback_@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:234:17\nchain/<@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:195:15\npromise callback*chain@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:175:5\nshaka.net.NetworkingEngine.prototype.send_@https://shaka-player-demo.appspot.com/lib/net/networking_engine.js:467:23\nshaka.net.NetworkingEngine.prototype.send_/sendOperation<@https://shaka-player-demo.appspot.com/lib/net/networking_engine.js:507:14\nwrapChainCallback_@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:234:17\nchain/<@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:208:15\npromise callback*chain@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:175:5\nshaka.net.NetworkingEngine.prototype.send_@https://shaka-player-demo.appspot.com/lib/net/networking_engine.js:467:23\nshaka.net.NetworkingEngine.prototype.makeRequestWithRetry_@https://shaka-player-demo.appspot.com/lib/net/networking_engine.js:416:10\nshaka.net.NetworkingEngine.prototype.request/requestOperation<@https://shaka-player-demo.appspot.com/lib/net/networking_engine.js:329:13\nwrapChainCallback_@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:234:17\nchain/<@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:195:15\npromise callback*chain@https://shaka-player-demo.appspot.com/lib/util/abortable_operation.js:175:5\nshaka.net.NetworkingEngine.prototype.request@https://shaka-player-demo.appspot.com/lib/net/networking_engine.js:328:26\nshaka.media.StreamingEngine.prototype.fetch_@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:2151:12\nshaka.media.StreamingEngine.prototype.initSourceBuffer_@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:1704:19\nshaka.media.StreamingEngine.prototype.fetchAndAppend_@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:1494:26\nshaka.media.StreamingEngine.prototype.update_@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:1246:3\nshaka.media.StreamingEngine.prototype.onUpdate_@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:1102:17\nsetTimeout handler*shaka.media.StreamingEngine.prototype.scheduleUpdate_@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:2208:28\nshaka.media.StreamingEngine.prototype.initStreams_/<@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:884:9\npromise callback*shaka.media.StreamingEngine.prototype.initStreams_@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:864:10\nshaka.media.StreamingEngine.prototype.init@https://shaka-player-demo.appspot.com/lib/media/streaming_engine.js:418:10\nshaka.Player.prototype.load@https://shaka-player-demo.appspot.com/lib/player.js:779:11\nAsync*shakaDemo.load/<@https://shaka-player-demo.appspot.com/demo/asset_section.js:313:12\npromise callback*shakaDemo.load@https://shaka-player-demo.appspot.com/demo/asset_section.js:311:3\nEventListener.handleEvent*shakaDemo.setupAssets_@https://shaka-player-demo.appspot.com/demo/asset_section.js:111:3\nshakaDemo.init/<@https://shaka-player-demo.appspot.com/demo/main.js:203:24\npromise callback*shakaDemo.init@https://shaka-player-demo.appspot.com/demo/main.js:185:5\nEventListener.handleEvent*@https://shaka-player-demo.appspot.com/demo/main.js:742:5\n" }
main.js:687:3
`

**Comment by joeyparrish** [comment]
Since #1498 is easy for me to reproduce, I'm marking this one as a duplicate of that (even though this issue was filed first).

Work on this issue will continue on #1498, and I'm hopeful that we can find a workaround at the very least.
